consensus/src/sync/live/state_queue/mod.rs

```diff
@@ -568,8 +577,23 @@ impl<N: Network> Stream for StateQueue<N> {
             }
         }
 
+        // Check if we have peers.
+        if self.nonempty_peers.is_none() {
```

Comment:
Isn't it better to use `get_or_insert`?

Reply:
`notify_nonempty_peers()` can return `None` thus the following assignment can be to `None` again, which `get_or_insert` does not allow for.